src/main/java/com/thebois/models/beings/pathfinding/TileCostComparator.java

```diff
@@ -0,0 +1,27 @@
+package com.thebois.models.beings.pathfinding;
+
+import java.util.Comparator;
+
+import com.thebois.models.world.ITile;
+
+/**
+ * Compares two tiles to determine which costs more to move to.
+ */
+class TileCostComparator implements Comparator<ITile> {
+
+    private final TileCostCalculator costCalculator;
+    private final ITile destination;
+
+    TileCostComparator(final TileCostCalculator costCalculator, final ITile destination) {
+        this.costCalculator = costCalculator;
+        this.destination = destination;
+    }
+
+    @Override
+    public int compare(final ITile first, final ITile second) {
+        final float firstCost = costCalculator.costOf(first, destination);
+        final float secondCost = costCalculator.costOf(second, destination);
+        return Float.compare(firstCost, secondCost);
```

Comment:
Thing is, comparators are a specific construct in Java. They are used by collections like the `PriorityQueue` (used in the `AstarPathFinder`) to sort elements. It has to be in a separate class, because it's implementing `Comparator<ITile>`. An anonymous inner class would be even worse, wouldn't it?

Reply:
Maybe you missed the comment:
```java
// The priority queue automatically sorts elements according to the comparator.
this.openSet = new PriorityQueue<>(costComparator);
```